### PR TITLE
Boards STM32: add SPI support on Arduino standard spi pin whenever possible

### PR DESCRIPTION
**Pull request by ABOSTM**
Boards STM32: add SPI support on Arduino standard spi pin (D10, D11, D12 and D13) whenever possible.

When necessary (when spi_nss pin not available on D10), use cs-gpio for NSS signal.

### CODE REVIEW

File: boards/arm/nucleo_l433rc_p/nucleo_l433rc_p.dts [hunk 1]
```diff
@@ -82,6 +82,13 @@
 	status = "okay";
 };
 
+
```
Comment (tagunil) [inline boards/arm/nucleo_l433rc_p/nucleo_l433rc_p.dts:85]: Excessive blank line here.

File: boards/shields/x_nucleo_idb05a1/boards/nucleo_l476rg.overlay [hunk 2]
```diff
@@ -0,0 +1,10 @@
+/*
+ * Copyright (c) 2019 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
```
Comment (tagunil) [inline boards/shields/x_nucleo_idb05a1/boards/nucleo_l476rg.overlay:6]: Excessive blank line here.

File: boards/arm/nucleo_l433rc_p/nucleo_l433rc_p.dts (hunk 1 above)
Comment (ABOSTM) [reply boards/arm/nucleo_l433rc_p/nucleo_l433rc_p.dts]: Thanks. Fixed.

File: boards/shields/x_nucleo_idb05a1/boards/nucleo_l476rg.overlay (hunk 2 above)
Comment (ABOSTM) [reply boards/shields/x_nucleo_idb05a1/boards/nucleo_l476rg.overlay:6]: Thanks. Fixed

**Comment by ABOSTM** [comment]
The remaining buildkite issue is: 
https://github.com/zephyrproject-rtos/zephyr/issues/36448

**Comment by ABOSTM** [comment]
Rebased to benefit from merge of buildkite fix (#36454)

**Review by erwango — APPROVED** [review]

**Review by FRASTM — APPROVED** [review]

**Pull request merged**